spirv_msl.cpp

```diff
@@ -1484,13 +1521,38 @@ uint32_t CompilerMSL::get_target_components_for_fragment_location(uint32_t locat
 uint32_t CompilerMSL::build_extended_vector_type(uint32_t type_id, uint32_t components, SPIRType::BaseType basetype)
 {
 	uint32_t new_type_id = ir.increase_bound_by(1);
-	auto &type = set<SPIRType>(new_type_id, get<SPIRType>(type_id));
-	type.vecsize = components;
+	auto &old_type = get<SPIRType>(type_id);
+	auto *type = &set<SPIRType>(new_type_id, old_type);
+	type->vecsize = components;
 	if (basetype != SPIRType::Unknown)
-		type.basetype = basetype;
-	type.self = new_type_id;
-	type.parent_type = type_id;
-	type.pointer = false;
+		type->basetype = basetype;
+	type->self = new_type_id;
+	type->parent_type = type_id;
+	type->array.clear();
+	type->array_size_literal.clear();
+	type->pointer = false;
+
+	if (is_array(old_type))
+	{
+		uint32_t array_type_id = ir.increase_bound_by(1);
+		type = &set<SPIRType>(array_type_id, *type);
+		type->parent_type = new_type_id;
+		std::copy(old_type.array.begin(), old_type.array.end(), type->array.begin());
```

Comment:
This is sketchy, what's the size of `type->array` here?

Reply:
Probably not the same as `old_type.array`. I changed it to an assign.